impl/maven-impl/src/main/java/org/apache/maven/impl/DefaultSourceRoot.java

```diff
@@ -65,24 +65,26 @@ public final class DefaultSourceRoot implements SourceRoot {
      * @param source a source element from the model
      */
     public DefaultSourceRoot(final Session session, final Path baseDir, final Source source) {
-        String value = nonBlank(source.getDirectory());
-        if (value == null) {
-            throw new IllegalArgumentException("Source declaration without directory value.");
-        }
-        directory = baseDir.resolve(value);
-        FileSystem fs = directory.getFileSystem();
+        FileSystem fs = baseDir.getFileSystem();
         includes = matchers(fs, source.getIncludes());
         excludes = matchers(fs, source.getExcludes());
         stringFiltering = source.isStringFiltering();
         enabled = source.isEnabled();
         moduleName = nonBlank(source.getModule());
 
-        value = nonBlank(source.getScope());
+        String value = nonBlank(source.getScope());
         scope = (value != null) ? session.requireProjectScope(value) : ProjectScope.MAIN;
 
         value = nonBlank(source.getLang());
```

Comment:
We should not be reusing these local variables for different things

Reply:
This is often true. But in this particular case, I have some difficulties to find a nice alternative. Multiplying the number of variables also increases the risk that we use the wrong ones. In the current code, the `value` variable is used each time for initializing a final field, then immediately assigned the next value, with each block separated by a blank line. There is no much room for confusion.

Reply:
It's not new in this PR so it's not a blocker, but I don't agree this is OK and I wouldn't have approved it if this were new code. Each could simply be named based on the source field it's derived from

Reply:
Or we need to relax the fact that we disallow inner blocks.  Those provide a scoping for local variables, so I don't think we should forbid both inner scopes _and_ reusing variables.

Reply:
> Or we need to relax the fact that we disallow inner blocks.

Yes exactly. I thought about relaxing that rule just for this method with a `@SuppressWarnings("checkstyle:...")`, but thought the maybe the community wanted strict adherence to the checkstyle rules.

Reply:
I'm fine with local relaxation.